core/commands/files/files.go

```diff
@@ -0,0 +1,555 @@
+		ShortDescription: `
+Read a specified number of bytes from a file at a given offset. By default, will
+read the entire file similar to unix cat.
+
+Examples:
+
+    $ ipfs files read /test/hello
+    hello
+		`,
+	},
+
+	Arguments: []cmds.Argument{
+		cmds.StringArg("path", true, false, "path to file to be read"),
+	},
+	Options: []cmds.Option{
+		cmds.IntOption("o", "offset", "offset to read from"),
+		cmds.IntOption("n", "count", "maximum number of bytes to read"),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		path := req.Arguments()[0]
+		fsn, err := mfs.Lookup(n.FilesRoot, path)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		fi, ok := fsn.(*mfs.File)
+		if !ok {
+			res.SetError(fmt.Errorf("%s was not a file", path), cmds.ErrNormal)
+			return
+		}
+
+		offset, _, _ := req.Option("offset").Int()
+
+		_, err = fi.Seek(int64(offset), os.SEEK_SET)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		var r io.Reader = fi
+		count, found, err := req.Option("count").Int()
+		if err == nil && found {
+			r = io.LimitReader(fi, int64(count))
+		}
+
+		res.SetOutput(r)
+	},
+}
+
+var FilesMvCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Move files",
+		ShortDescription: `
+Move files around. Just like traditional unix mv.
+
+Example:
+
+    $ ipfs files mv /myfs/a/b/c /myfs/foo/newc
+
+		`,
+	},
+
+	Arguments: []cmds.Argument{
+		cmds.StringArg("source", true, false, "source file to move"),
+		cmds.StringArg("dest", true, false, "target path for file to be moved to"),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		src := req.Arguments()[0]
+		dst := req.Arguments()[1]
+
+		err = mfs.Mv(n.FilesRoot, src, dst)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+	},
+}
+
+var FilesWriteCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Write to a mutable file in a given filesystem",
+		ShortDescription: `
+Write data to a file in a given filesystem. This command allows you to specify
+a beginning offset to write to. The entire length of the input will be written.
+
+If the '--create' option is specified, the file will be create if it does not
+exist. Nonexistant intermediate directories will not be created.
+
+Example:
+
+	echo "hello world" | ipfs files write --create /myfs/a/b/file
+	echo "hello world" | ipfs files write --truncate /myfs/a/b/file
+		`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("path", true, false, "path to write to"),
+		cmds.FileArg("data", true, false, "data to write").EnableStdin(),
+	},
+	Options: []cmds.Option{
+		cmds.IntOption("o", "offset", "offset to write to"),
+		cmds.BoolOption("n", "create", "create the file if it does not exist"),
```

Comment:
actually: 
- would be good to have a `-n, --count` option to specify how many bytes from stdin the read (allowing you to not read everything).
- maybe `-e` for `--create` ?